01-node-tutorial/11-fs-async.js

```diff
@@ -1,29 +1,40 @@
-const { readFile, writeFile } = require('fs')
+const { writeFile } = require('fs');
 
-console.log('start')
-readFile('./content/first.txt', 'utf8', (err, result) => {
+console.log("at start");
+// Nesting Callback functions required to ensure operations happen in the correct order
+writeFile(
+  './answers/temporary/fileB.txt', 
+  'This is line 1\n', 
+  (err) => {
   if (err) {
-    console.log(err)
-    return
-  }
-  const first = result
-  readFile('./content/second.txt', 'utf8', (err, result) => {
-    if (err) {
-      console.log(err)
-      return
-    }
-    const second = result
+    console.log("This error happened: ", err);
+  } else {
+    console.log("Line 1");
+
     writeFile(
-      './content/result-async.txt',
-      `Here is the result : ${first}, ${second}`,
-      (err, result) => {
-        if (err) {
-          console.log(err)
-          return
-        }
-        console.log('done with this task')
+      './answers/temporary/fileB.txt', 
+      'This is line 2\n', 
+      { flag: 'a' }, 
+      (err) => {
+      if (err) {
+        console.log("This error happened: ", err);
+      } else {
+        console.log("Line 2");
+
+        writeFile(
+          './answers/temporary/fileB.txt', 
+          'This is line 3\n', 
+          { flag: 'a' }, 
+          (err) => {
+          if (err) {
+            console.log("This error happened: ", err);
```

Comment:
make sure to include the 'return' statement in the error handler, so that the function stops at the error.  good otherwise.